nrf5/hal/hal_rtc.c

```diff
@@ -32,84 +32,85 @@
 
 static hal_rtc_app_callback m_callback;
 
+static uint32_t m_period[sizeof(RTC_BASE_POINTERS) / sizeof(uint32_t)];
+
 void hal_rtc_callback_set(hal_rtc_app_callback callback) {
     m_callback = callback;
 }
 
 void hal_rtc_init(hal_rtc_conf_t const * p_rtc_conf) {
-    p_rtc_conf->p_instance->PRESCALER = (32768 / p_rtc_conf->frequency) - 1; // approx correct.
-    hal_irq_priority(p_rtc_conf->irq_num, p_rtc_conf->irq_priority);
+    NRF_RTC_Type * p_rtc = RTC_BASE(p_rtc_conf->id);
 
     // start LFCLK if not already started
     if (NRF_CLOCK->LFCLKSTAT == 0) {
         NRF_CLOCK->TASKS_LFCLKSTART = 1;
         while (NRF_CLOCK->EVENTS_LFCLKSTARTED == 0);
         NRF_CLOCK->EVENTS_LFCLKSTARTED = 0;
     }
+
+    m_period[p_rtc_conf->id] = p_rtc_conf->period;
+
+    p_rtc->PRESCALER = (32768 / 32) - 1; // approx ms ticks.
```

Comment:
Fixed. Updated also RTC to be configured with 10hz prescaler value. 1 hz did not work out properly, so its using a 10 times more fine grained tick count to resolve 1 second ticks.